library/general/src/lib/yast2/popup.rb

```diff
@@ -0,0 +1,318 @@
+        content_res = content(body(headline, message, richtext, timeout), buttons)
+
+        event_loop(content_res, focus, timeout, details, style)
+      end
+
+      # Shows feedback till given block does not finish.
+      # @param message [String] message to show
+      # @param headline [String] sets popup headline. String is shown.
+      #   If empty string is passed no headline is shown.
+      # @yield [] operation that needs to show feedback
+      # @return result of block
+      def feedback(message, headline: "", &block)
+        headline = generate_headline(headline)
+        if !message.is_a?(::String)
+          raise ArgumentError, "Invalid value #{message.inspect} of parameter message"
+        end
+
+        body = VBox(
+          VSpacing(0.4),
+          *headline_widgets(headline),
+          Left(Label(message))
+        )
+        content_res = content(body, {})
+
+        res = Yast::UI.OpenDialog(content_res)
+        raise "Failed to open dialog, see logs." unless res
+        begin
+          block.call
+        ensure
+          Yast::UI.CloseDialog
+        end
+      end
+
+    private
+
+      include Yast::UIShortcuts
+
+      def check_arguments!(message, details, timeout, focus, buttons)
+        if !message.is_a?(::String)
+          raise ArgumentError, "Invalid value #{message.inspect} of parameter message"
+        end
+
+        if !details.is_a?(::String)
+          raise ArgumentError, "Invalid value #{details.inspect} of parameter details"
+        end
+
+        if !timeout.is_a?(::Integer)
+          raise ArgumentError, "Invalid value #{timeout.inspect} of parameter timeout"
+        end
+
+        if !buttons.key?(focus)
+          raise ArgumentError, "Invalid value #{focus.inspect} for parameter focus. " \
+           "Known buttons: #{buttons.keys}."
+        end
+
+        nil
+      end
+
+      def generate_buttons(buttons)
+        case buttons
+        when ::Hash
+          buttons
+        when :ok
+          { ok: Yast::Label.OKButton }
+        when :continue_cancel
+          { continue: Yast::Label.ContinueButton, cancel: Yast::Label.CancelButton }
+        when :yes_no
+          { yes: Yast::Label.YesButton, no: Yast::Label.NoButton }
+        else
+          raise ArgumentError, "Invalid value #{buttons.inspect} for parameter buttons."
+        end
+      end
+
+      def generate_headline(headline)
+        case headline
+        when ::String
+          headline
+        when :warning
+          Yast::Label.WarningMsg
+        when :error
+          Yast::Label.ErrorMsg
+        else
+          raise ArgumentError, "Invalid value #{headline.inspect} for parameter headline."
+        end
+      end
+
+      def add_details_button(buttons)
+        # use this way merge to have details as first place button
+        { details: _("&Details...") }.merge(buttons)
+      end
+
+      def add_stop_button(buttons)
+        # use this way merge to have details as first place button
```

Comment:
Comment copy-paste.

And, in fact, to have `:stop` at the end you only need to do an straightforward `buttons[:stop] = Yast::Label.StopButton`

Reply:
make sense and I plan to modify even that details as there is a bug in ButtonBox